### PR TITLE
Fix invalid share perms

### PR DESCRIPTION
**Pull request by LukasReschke**
From https://github.com/owncloud/core/pull/26562

### CODE REVIEW

File: lib/private/Repair/RepairInvalidShares.php [hunk 1]
```diff
@@ -92,6 +93,35 @@ private function addShareLinkDeletePermission(IOutput $out) {
...
+	 * Adjust file share permissions
+	 */
+	private function adjustFileSharePermissions(IOutput $out) {
+		$mask = \OCP\Constants::PERMISSION_READ | \OCP\Constants::PERMISSION_UPDATE | \OCP\Constants::PERMISSION_SHARE;
+		$builder = $this->connection->getQueryBuilder();
+
+
+		if ($this->connection->getDatabasePlatform() instanceof OraclePlatform) {
+			$permsFunc = $builder->createFunction(
+				'bitand(' . $builder->getColumnName('permissions') . ', ' . $mask . ')'
```
Comment (nickvergessen) [inline lib/private/Repair/RepairInvalidShares.php]: > But I guess to correctly fix this, you should add a new method to IExpressionBuilder. Then Oracle can haz it's own fucked up implementation.

My comment from the OC PR stands, I think we should add that as a method to the expression builder.
I can also do that after my vacation myself.

**Review by nickvergessen — DISMISSED** [review]

File: lib/private/Repair/RepairInvalidShares.php (hunk 1 above)
Comment (LukasReschke) [inline lib/private/Repair/RepairInvalidShares.php]: > I can also do that after my vacation myself.

I'd appreciate that :)

**Comment by nickvergessen** [comment]
Updated, please have a look @LukasReschke @rullzer

**Comment by rullzer** [comment]
@nickvergessen failing tests
I would just let share.php be for now until it dies off.

**Review by rullzer — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**